dask/bag/core.py

```diff
@@ -1538,6 +1538,10 @@ def groupby(
         if method is not None:
             raise Exception("The method= keyword has been moved to shuffle=")
         if shuffle is None:
+            if not isinstance(grouper, Callable):
+                # tasks doesn't work with non-callables (which shouldn't be supported
```

Comment:
Checking in, why shouldn't this be supported?

Reply:
the docstring states that it's a callable, no alternatives.

Reply:
An alternative to removing the functionality would be to change the docstring.

I think the question is if this functionality should be supported or not.  If we think it isn't then we would probaby deprecate it.  If we think that it is then we would probaby change the docstring.